packages/engine/src/simulation/task/mod.rs

```diff
@@ -103,7 +103,7 @@ use crate::simulation::{
 // Additionally we have TryInto<init::Task>, (and others)
 // implemented for this enum.
 #[derive(Clone, Debug)]
-pub enum Task {
+pub enum PackageTask {
```

Comment:
I'm not sure I like this change, (also the docs still need changing). I'm just not sure it has clearer meaning, I don't see these as being "Package" tasks. They're tasks, Packages happen to be able to spawn them.

Happy to hear the justification for this

Reply:
This was mainly done to avoid confusion between `enum Task` and `trait Task`. If we decide to revert the changes for `trait Task`, I'm happy to revert this as well.
Additionally, this was done because the variants are specifically for the different packages, so the tasks are very specific.

Reply:
I think I'm okay with it as is, at least it's an improvement on how it was before I think. We can always revisit if it doesn't seem to make sense / is confusing later on